iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -297,23 +299,39 @@ struct ConvertHLOToLinalgOnTensorsPass
     MLIRContext *context = &getContext();
 
     auto typeConverter = mhlo::createHloToLinalgSignedIntegerConverter();
-    populateHLOToLinalgOnTensorsConversionPatterns(context, *typeConverter,
-                                                   patterns);
+    if (directHloClientLowering) {
+      // NOTE: not using corresponding setupHLOToFlowPatterns because the entire
+      // HLO dialects are marked illegal by this pass.
+      // TODO: Wonky.
```

Comment:
Could you outline what is wonky for the uninformed :P

Reply:
Lol, copy-pasta from my WIP branch. Forgot that was in there.